README.developer.md

```diff
@@ -0,0 +1,18 @@
+# NMState Coding Guidelines
+
+- NMState is written primarily in Python, and its coding style should follow the best practices of Python coding unless otherwise declared.
+- PEP8 is holy.
+- Tests are holy. Production code must be covered by unit tests and basic integration tests.
```

Comment:
We do not follow this completely, therefore "must" is a little bit too strong.

Reply:
I am OK with `must`. But it could be better we could say more about how to create unit test and integration test like:
```
Please refer to commit <commit_id_with_unit_test_integration_test> as an example.
```

Reply:
'must': We should follow it, and I hope you will nack my patches if it does not. But I should rephrase it to: "unit tests and/or basic integration tests". There are tests which are better covered using the integration tests and not unit tests (the ones who need too much mocking).

About the 'how': Well, I do hope the one who attempts to post a change knows where too look. I usually go to tests to learn how to change something or how it currently works.